data/data/openstack/main.tf

```diff
@@ -40,6 +40,7 @@ module "bootstrap" {
   master_port_ids         = module.topology.master_port_ids
   root_volume_size        = var.openstack_master_root_volume_size
   root_volume_type        = var.openstack_master_root_volume_type
+  zone                    = var.openstack_master_availability_zones[0]
```

Comment:
I wrongly assumed that setting the default like this would cover that:

```hcl
variable "openstack_master_availability_zones" {
  type = list(string)
  default = [""]
  description = "List of availability Zones to Schedule the masters on"
}
```

Reply:
see next comment

Reply:
Unless I'm mistaken, this could still break if someone explicitly sets the terraform variable, for example via environment variable. This isn't really a big issue, but the above code was safer in that regard.

Also, you don't have to do the trick of returning a slice with an empty string if there is no zone, it's a bit confusing.